### ISSUE TITLE
Splatting for native commands results in incompatible syntax for modern commands

### ISSUE THREAD

**Issue by SteveL-MSFT** [PowerShell/PowerShell#20068]
### Summary of the new feature / enhancement

Since PS5.1, splatting to a native commands results in a single `-` prefix and a `:` separator between the parameter name and the parameter value.

```powershell
$a = @{a=1;bb=2}
echoit @a
```

Expected where the convention is a single character parameter is prefixed by a single dash and a spelled out parameter is prefixed by two dashes and parameter values are space separated:

```
echoit -a 1 --bb 2
```

Actual:

```
echoit -a:1 -bb:2
```

The current behavior means this doesn't work with most modern commands.  If the hashtable is converted to an array, you can get some of the right behavior as splatting an array always means separate parameters and prefixes need to be defined within the array.

### Proposed technical implementation details (optional)

If we are ok with a breaking change as it could be a bucket 3 item, then we use the logic that single character parameters are prefixed by single dash while multi-character parameter names are prefixed by a double-dash.

If we need to support `-`, `--`, `/`, etc... explicitly as well as `:`,`=`, etc.. separator, then we could have a predefined special member of the hashtable that defines those options:

```powershell
$params = @{
  _prefix = '/'
  _separator = '='
  arg1 = 1
  arg2 = 2
}
```

I'm choosing the underscore so it minimizes conflicts with existing parameter names.  There is a discovery problem that can only be solved by docs or a feedbackprovider that sees splatting and the native command failing and pointing to the docs.

**Comment by JamesWTruher** [comment]
also this!
```powershell
PS> $ht = @{ one = 1; "-two" = 2 }
PS> echoit @ht                    
Argument 1 <-one:1>
Argument 2 <--two:2>
```

**Comment by SteveL-MSFT** [comment]
Yes, if a prefix is already supplied, we shouldn't add any additional one.  And if there is a separator as part of the parameter name, it also shouldn't add it.

**Comment by mklement0** [comment]
@SteveL-MSFT,  while the convention you're referring to - a `-` prefix for _single-character_ option names vs. `--` for _multi-character_ option names - applies to _GNU_  (Linux) CLIs and many, but not all (see below) macOS CLIs, this convention isn't _universally_ observed; e.g.:

* `java` accepts a _mix_ of `-`- and `--`-prefixed multi-character options, depending on the specific option used; e.g. `java -cp ...` works, but `java --cp ...` doesn't.

* On macOS, there are NeXTStep-derived utilities such as `default` and `mdfind`, whose options are (a) _all_ multi-character options and (b) _never_ accept a `--` prefix.

The short of it:

* Adopting the GNU convention for _hashtable_-based splatting is a sensible improvement, but its limitation should be clearly documented.

* **[about_Splatting](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_Splatting) should clearly document that for _native programs_ it's best to use _array_-based splatting, which gives the user full control over whether `-`- or `--`-prefixing is required**; e.g.:

  ```powershell
  $a = '-a', 1, '--b', 2 # use an *array* to control how option names are passed
  echoit @a
  ```
